src/main/java/jcifs/smb/SmbSessionImpl.java

```diff
@@ -0,0 +1,1297 @@
+        }
+
+        try {
+            final String hostName = host;
+            return Subject.doAs(s, new PrivilegedExceptionAction<SSPContext>() {
+
+                @Override
+                public SSPContext run () throws Exception {
+                    return getCredentials().createContext(getContext(), tdomain, hostName, negoResp.getSecurityBlob(), doSigning);
+                }
+
+            });
+        }
+        catch ( PrivilegedActionException e ) {
+            if ( e.getException() instanceof SmbException ) {
+                throw (SmbException) e.getException();
+            }
+            throw new SmbException("Unexpected exception during context initialization", e);
+        }
+    }
+
+
+    /**
+     * @param request
+     * @param response
+     * @param params
+     * @return
+     * @throws CIFSException
+     */
+    @SuppressWarnings ( "unchecked" )
+    private <T extends CommonServerMessageBlock> T reauthenticate ( SmbTransportImpl trans, final String tdomain,
+            CommonServerMessageBlockRequest chain, T andxResponse, Set<RequestParam> params ) throws CIFSException {
+        SmbException ex = null;
+        Smb2SessionSetupResponse response = null;
+        Smb2NegotiateResponse negoResp = (Smb2NegotiateResponse) trans.getNegotiateResponse();
+        byte[] token = negoResp.getSecurityBlob();
+        final int securityMode = negoResp.getSecurityMode();
+        boolean anonymous = this.credentials.isAnonymous();
+        final boolean doSigning = securityMode != 0 && !anonymous;
+        long newSessId = 0;
+        long curSessId = this.sessionId;
+
+        synchronized ( trans ) {
+            this.credentials.refresh();
+            Subject s = this.credentials.getSubject();
+            SSPContext ctx = createContext(trans, tdomain, negoResp, doSigning, s);
+            while ( true ) {
+                token = createToken(ctx, token, s);
+
+                if ( token != null ) {
+                    Smb2SessionSetupRequest request = new Smb2SessionSetupRequest(
+                        getContext(),
+                        negoResp.getSecurityMode(),
+                        negoResp.getCommonCapabilities(),
+                        curSessId,
+                        token);
+
+                    if ( chain != null ) {
+                        request.chain((ServerMessageBlock2) chain);
+                    }
+
+                    request.setDigest(this.digest);
+                    request.setSessionId(curSessId);
+
+                    try {
+                        response = trans.send(request, null, EnumSet.of(RequestParam.RETAIN_PAYLOAD));
+                        newSessId = response.getSessionId();
+
+                        if ( newSessId != curSessId ) {
+                            throw new SmbAuthException("Server did not reauthenticate after expiration");
+                        }
+                    }
+                    catch ( SmbAuthException sae ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception is caught and re-thrown within the catch clause without any other operations being performed on the exception, making the catch clause unnecessary.

**Fix**: Perform additional operations on the exception that has been caught. You can wrap a standard exception into a custom exception to add additional information or special handling. When you do that, make sure to set the original exception as the cause. Otherwise, you will lose the stack trace and message of the original exception, which will make it difficult to analyze the event that caused the exception.  
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)